Menus/BuyMenuGUI.cpp

```diff
@@ -1208,20 +1208,45 @@ void BuyMenuGUI::Update()
-        string description = "";
-        // Get it from the regular Entitiy preset
-        if (pItem && pItem->m_pEntity && !pItem->m_pEntity->GetDescription().empty())
+        if (pItem && pItem->m_pEntity) {
             description = pItem->m_pEntity->GetDescription();
-        // Show popup info box next to selected module item, but only if it has a description
-        else if (pItem && pItem->m_ExtraIndex >= 0)
-        {
-            const DataModule *pModule = g_PresetMan.GetDataModule(pItem->m_ExtraIndex);
-            if (pModule && !pModule->GetDescription().empty())
+            const Entity* currentItem = pItem->m_pEntity;
+            const ACraft* pItemAsCraft = dynamic_cast<const ACraft*>(currentItem);
+            if (pItemAsCraft) {
+                int craftMaxPassengers = pItemAsCraft->GetMaxPassengers();
+                int craftMaxMass = static_cast<int>(pItemAsCraft->GetMaxMass()) - static_cast<int>(pItemAsCraft->GetMass());
+                description += "\nMax Mass: " + std::to_string(craftMaxMass) + "\nMax Passengers: " + std::to_string(craftMaxPassengers);
+            } else {
+                std::stringstream floatStream;
+                const Actor* pItemAsActor = dynamic_cast<const Actor*>(currentItem);
+                if (pItemAsActor) {
+                    floatStream << std::fixed << std::setprecision(1) << pItemAsActor->GetMass();
+                    // If the item's mass is an integer, remove trailing zero and decimal point.
+                    description += "\nMass: " + ((floatStream.str().back() == '0') ? floatStream.str().substr(0, floatStream.str().length() - 2) : floatStream.str());
+                    
+                    int passengerSlotsTaken = pItemAsActor->GetPassengerSlots();
+                    if (passengerSlotsTaken > 1) {
+                        description += "\nPassenger Slots: " + std::to_string(passengerSlotsTaken);
+                    }
+                }else {
+                    const MovableObject* pItemAsMO = dynamic_cast<const MovableObject*>(currentItem);
+                    if (pItemAsMO) {
+                        floatStream << std::fixed << std::setprecision(1) << pItemAsMO->GetMass();
+                        // If the item's mass is an integer, remove trailing zero and decimal point.
+                        description += "\nMass: " + ((floatStream.str().back() == '0') ? floatStream.str().substr(0, floatStream.str().length() - 2) : floatStream.str());
+                    }
+                }
+            }
+        }else if (pItem && pItem->m_ExtraIndex >= 0) {
+            const DataModule* pModule = g_PresetMan.GetDataModule(pItem->m_ExtraIndex);
+            if (pModule && !pModule->GetDescription().empty()) {
                 description = pModule->GetDescription();
+            }
         }
 
+        // Show popup info box next to selected item, but only if it has a description
```

Comment:
I think this is junk?

Reply:
This was a legacy comment, I'll remove it.